File: snippets/curves/linear.js
// Return the polynomial value and its derivative at x.
function evaluate_linear(x, a, b) {
  if (![x, a, b].every(Number.isFinite)) throw new RangeError("Use finite inputs.");
  const value = a * x + b;
  const derivative = a;
  if (![value, derivative].every(Number.isFinite)) throw new RangeError("Result overflow.");
  return { value, derivative };
}

File: snippets/curves/linear.py
import math


def evaluate_linear(x, a, b):
    """Return the polynomial value and its derivative at x."""
    if not all(math.isfinite(n) for n in [x, a, b]):
        raise ValueError("Use finite inputs.")
    value = a * x + b
    derivative = a
    if not all(math.isfinite(n) for n in [value, derivative]):
        raise ValueError("Result overflow.")
    return {"value": value, "derivative": derivative}
